### PR TITLE
rails7 and ruby3

### PR DESCRIPTION
**Pull request by matafc**
Details

- Upgrade to rails 7
- Upgrade to ruby 3.2.2
- Remove debase and ruby-debug-ide due to https://github.com/ruby-debug/debase/issues/76, fixing that can be in scope of another PR if anyone wants it
- Some syntax and config changes due to updated dependencies
- hosts.config needs to be defined now
- updated docker-compose mounts to make iteration easier without have to rebuild the images

This was tested locally with the [e2e](https://github.com/degica/barcelona-e2e) test.

### CODE REVIEW

**Comment by degikko** [comment]
@essa can you help us review this PR, please?

File: config/environments/production.rb [hunk 1]
```diff
@@ -83,4 +83,7 @@
 
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
+
+  config.hosts << "barcelona.degica.com"
+  
 end
```
Comment (essa) [inline config/environments/production.rb:89]: I think you have to exclude a path for ALB health check.
Similar issue is [here](https://github.com/degica/hcs/pull/501).

**Review by essa — CHANGES_REQUESTED** [review]
Almost perfect besides one issue about ALB health check.
See my comment for detail

File: config/environments/production.rb (hunk 1 above)
Comment (matafc) [reply config/environments/production.rb:91]: Thank you for that! I was worried I was missing something else. Updated to include the fix.

**Pull request merged**